jerry-core/ecma/base/ecma-helpers-value.c

```diff
@@ -275,21 +300,146 @@ ecma_check_value_type_is_spec_defined (ecma_value_t value) /**< ecma value */
+ *   The value must fit into the range of allowed ecma integer values
+ *
+ * @return ecma-value
  */
-ecma_value_t __attr_const___
-ecma_make_number_value (const ecma_number_t *ecma_num_p) /**< number to reference in value */
+inline ecma_value_t __attr_const___ __attr_always_inline___
+ecma_make_integer_value (ecma_value_integer_t value) /**< int32 number to be encoded */
+{
+  JERRY_ASSERT (value <= ECMA_INTEGER_NUMBER_MAX
+                && value >= ECMA_INTEGER_NUMBER_MIN);
+
+  return (ecma_value_t) ((value << ECMA_INTEGER_NUMBER_SHIFT) | ECMA_TYPE_NON_PTR_VALUE);
+} /* ecma_make_integer_value */
+
+/**
+ * Allocate and initialize a new float number without checks.
+ *
+ * @return ecma-value
+ */
+static ecma_value_t __attr_const___
+ecma_create_float_number (ecma_number_t ecma_num) /**< value of the float number */
 {
-  JERRY_ASSERT (ecma_num_p != NULL);
+  ecma_number_t *ecma_num_p = ecma_alloc_number ();
+
+  *ecma_num_p = ecma_num;
+
+  return ecma_pointer_to_ecma_value (ecma_num_p) | ECMA_TYPE_FLOAT_NUMBER;
+} /* ecma_create_float_number */
+
+/**
+ * Create a new float number (usually from NaN).
+ *
+ * @return ecma-value
+ */
+inline ecma_value_t __attr_always_inline___
+ecma_make_float_number_value (ecma_number_t ecma_num) /**< value of the float number */
+{
+  JERRY_ASSERT (((ecma_number_t) (ecma_value_integer_t) ecma_num) != ecma_num);
+
+  return ecma_create_float_number (ecma_num);
+} /* ecma_make_float_number_value */
+
+static inline bool __attr_const___ __attr_always_inline___
+ecma_is_number_equals_to_positive_zero (ecma_number_t ecma_num) /**< number */
+{
+#if CONFIG_ECMA_NUMBER_TYPE == CONFIG_ECMA_NUMBER_FLOAT32
+  union
+  {
+    uint32_t u32_value;
+    ecma_number_t float_value;
+  } u;
+
+  u.float_value = ecma_num;
+
+  return u.u32_value == 0;
+#else
+  union
+  {
+    uint64_t u64_value;
+    ecma_number_t float_value;
+  } u;
 
-  return ecma_pointer_to_ecma_value (ecma_num_p) | ECMA_TYPE_NUMBER;
+  u.float_value = ecma_num;
+
+  return u.u64_value == 0;
+#endif
+}
+
+/**
+ * Encode a number into an ecma-value
+ *
+ * @return ecma-value
+ */
+ecma_value_t
+ecma_make_number_value (ecma_number_t ecma_num) /**< number to be encoded */
+{
+  ecma_value_integer_t integer_value = (ecma_value_integer_t) ecma_num;
+
+  if ((ecma_number_t) integer_value == ecma_num)
+  {
+    if (integer_value == 0)
+    {
+      if (ecma_is_number_equals_to_positive_zero (ecma_num))
+      {
+        return ecma_make_integer_value (0);
+      }
+    }
+    else if (integer_value <= ECMA_INTEGER_NUMBER_MAX
+             && integer_value >= ECMA_INTEGER_NUMBER_MIN)
+    {
+      return ecma_make_integer_value (integer_value);
+    }
+  }
+
+  return ecma_create_float_number (ecma_num);
 } /* ecma_make_number_value */
 
 /**
+ * Encode an int32 number into an ecma-value
+ *
+ * @return ecma-value
+ */
+ecma_value_t
+ecma_make_int32_value (int32_t ecma_num) /**< int32 number to be encoded */
+{
+  if (ecma_num <= ECMA_INTEGER_NUMBER_MAX
```

Comment:
Could you write like this:
```C++
if (ECMA_INTEGER_NUMBER_MIN <= ecma_num
    && ecma_num <= ECMA_INTEGER_NUMBER_MAX
```
It's better to read because it's more alike `min <= X <= max`